File: frontend/src/app/model/restaurantOrderDTO.ts
import { Location } from './location'
import { FoodQuantityDTO } from './foodQuantityDTO'


export interface RestaurantOrderDTO {
    id: number
    dateCreated: string
    status: string
    location: Location
    customerId: number
    customerName: string
    customerSurname: string
    customerUsername: string
    paymentMethod: string
    cartId: number
    foodQuantity: FoodQuantityDTO[]
  }

File: backend/src/main/kotlin/com/sorsix/backend/model/dto/RestaurantOrderDTO.kt
package com.sorsix.backend.model.dto

import com.sorsix.backend.model.*
import com.sorsix.backend.model.enumeration.OrderStatus
import com.sorsix.backend.model.enumeration.PaymentMethod
import java.time.LocalDate

data class RestaurantOrderDTO(
    val id: Long? = null,
    val dateCreated: LocalDate,
    var status: OrderStatus,
    var location: Location,
    var customerId: Long,
    var customerName: String,
    var customerSurname: String,
    var customerUsername: String,
    var paymentMethod: PaymentMethod,
    var cartId: Long? = null,
    var foodQuantity: List<FoodNameQuantityDTO>

)